src/nvim/api/buffer.c

```diff
@@ -1207,7 +1207,57 @@ Integer nvim_buf_set_virtual_text(Buffer buffer,
   return 0;
 }
 
-Dictionary nvim__buf_stats(Buffer buffer, Error *err)
+/// Get the virtual text (annotation) for a buffer line.
+///
+/// The virtual text is returned as list of lists, whereas the inner lists have
```

Comment:
We (or at least I) almost always regret List interfaces instead of Map, since Maps are easier to extend later. And this nested/ordered-pair scheme already strongly suggests this should be a map.

Also if `lnum=-1` should it return all annotations in the buffer? Keyed on line number. E.g.:

```
{
  42: { 'text': ..., 'hl_group': ... },
  109: { 'text': ..., 'hl_group': ... },
}
```